### PR TITLE
NAS-128645 / 24.10 / Add SSSD to pam configuration

### PR DESCRIPTION
**Pull request by anodos325**
This commit adds basic SSSD integration with our existing pam modules. Based on sample pam config provided by debian.

### CODE REVIEW

File: src/middlewared/middlewared/etc_files/pam.d/pam.inc.mako [hunk 1]
```diff
@@ -176,77 +172,45 @@
...
-            if self.is_kerberized():
-                krb5_entry = f"account\t\trequired\t{self.pam_krb5}\t\t{' '.join(krb5_args)}"
-                entries.insert(1, krb5_entry)
-
-            return {'primary': [unix_entry], 'additional': entries}
+            return {'primary': [unix_entry], 'additional': [ldap_entry]}
 
         def pam_session(self):
             entries = [super().pam_session()]
             entries.append(super().pam_session(pam_path=self.pam_ldap, pam_control='optional'))
```
Comment (mgrimesix) [inline src/middlewared/middlewared/etc_files/pam.d/pam.inc.mako:199]: This should be pam_sss

File: src/middlewared/middlewared/etc_files/pam.d/pam.inc.mako [hunk 2]
```diff
@@ -176,77 +172,45 @@
...
-                "ignore_unknown_user",
-                "ignore_authinfo_unavail",
-                "no_warn",
-                f"minimum_uid={min_uid}"
+                "use_authtok",
             ]
-            krb5_args = ["try_first_pass", f"minimum_uid={min_uid}"]
 
             unix_entry = super().pam_password(success=2)
             ldap_entry = super().pam_password(pam_path=self.pam_ldap, pam_args=ldap_args, success=1)
```
Comment (mgrimesix) [inline src/middlewared/middlewared/etc_files/pam.d/pam.inc.mako:210]: change to sssd entry?  Here and below.
e.g.
`sssd_entry = super().pam_password(pam_path=self.pam_sss, pam_args=ldap_args, success=1)`

File: src/middlewared/middlewared/etc_files/pam.d/pam.inc.mako [hunk 3]
```diff
@@ -176,77 +172,45 @@
...
 
             entries = [unix_entry, ldap_entry]
```
Comment (mgrimesix) [inline src/middlewared/middlewared/etc_files/pam.d/pam.inc.mako:212]: change to:
`entries = [unix_entry, sssd_entry]`

**Review by mgrimesix — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/etc_files/pam.d/pam.inc.mako [hunk 4]
```diff
@@ -3,9 +3,8 @@
         def __init__(self, **kwargs):
             self.middleware = kwargs.get('middleware')
             self.pam_mkhomedir = "pam_mkhomedir.so"
-            self.pam_ldap = "pam_ldap.so"
+            self.pam_ldap = "pam_sss.so"
```
Comment (mgrimesix) [inline src/middlewared/middlewared/etc_files/pam.d/pam.inc.mako:6]: Hmm... it means a little more changes, but I think we should not use pam_ldap for pam_sss.

**Review by mgrimesix — COMMENTED** [review]
Updated my review.

**Review by mgrimesix — APPROVED** [review]

**Review by bmeagherix — APPROVED** [review]

**Pull request merged**